apps/etherpad/templates/services/app/dc.yml.j2

```diff
@@ -45,16 +45,26 @@ spec:
                 --apikey "{{ etherpad_private_directory_path }}/APIKEY.txt"
           livenessProbe:
             httpGet:
-              path: /
-              port: {{ etherpad_application_port }}
-            initialDelaySeconds: 30
-            periodSeconds: 10
+              path: {{ etherpad_app_healthcheck_live_endpoint }}
+              port: {{ etherpad_app_healthcheck_port }}
+            failureThreshold: 3
+            initialDelaySeconds: 10
+            # Allow sufficient amount of time (90 seconds = periodSeconds * failureThreshold)
+            # for the registered shutdown handlers to run to completion.
+            periodSeconds: 30
+            successThreshold: 1
+            # Setting a very low timeout value (e.g. 1 second) can cause false-positive
+            # checks and service interruption.
+            timeoutSeconds: 5
           readinessProbe:
             httpGet:
-              path: /
-              port: {{ etherpad_application_port }}
+              path: {{ etherpad_app_healthcheck_ready_endpoint }}
+              port: {{ etherpad_app_healthcheck_port }}
+            failureThreshold: 1
```

Comment:
I find this value very low.
And since we have `replicas=1`, this leads to service interruption in case of false positive on the readiness check.

WDYT about setting `failureThreshold: 3` here ?

Reply:
I might be wrong, but this is a readiness probe, so there won't be any service interruption IMO since: i. in a rolling update scenario, the switch won't happen until the pod is ready, and, ii. in a blue-green scenario the deployment won't pursue if the pod is not ready (and the switch didn't happen yet).

FWIU, with a `failureThreshold` of 3, it means that we will need to wait for 3 failures to consider that the pod is not ready. Since we already know it the first time it occurs, why waiting for 2 other `periodsSeconds`?

Reply:
Oh, I didn't take in account the impact of the readiness probe during the deployment process while writing my comment.

From what I understand, the readiness probe has another impact at the service level  : if a pod is not ready, at anytime during its whole lifecycle, no traffic is sent to it by the service. That's why I said it generates an interruption.

Source : [K8s documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-readiness-probes) : 

> A pod with containers reporting that they are not ready does not receive traffic through Kubernetes Services.

and 

> Note: Readiness probes runs on the container during its whole lifecycle.

Reply:
I see. Thanks for the tip :pray: